### PR TITLE
[Model] Add RobertaForTokenClassification / XLMRobertaForTokenClassification

### PR DESCRIPTION
**Pull request by krishy91**
## Purpose

vLLM's model registry supports `BertForTokenClassification` and `ModernBertForTokenClassification` for the `/pooling` token-classify runner, but RoBERTa-family checkpoints have no token-classification entry. `RobertaForSequenceClassification`/`XLMRobertaForSequenceClassification` already exist, but a RoBERTa or XLM-RoBERTa checkpoint fine-tuned for token classification (e.g. NER) currently has no matching architecture and cannot be served correctly through vLLM's pooling runner.

This PR adds `RobertaForTokenClassification`, registered under both the `RobertaForTokenClassification` and `XLMRobertaForTokenClassification` architecture names (mirroring how `XLMRobertaForSequenceClassification` already aliases `RobertaForSequenceClassification`).

The new class is a direct structural mirror of the existing `BertForTokenClassification` (same `encoder_only` attention type, `ALL`-token pooling, per-token linear classifier head), swapping in `RobertaEmbedding` and the `roberta.*` weight prefix — the same adaptation vLLM already applies when deriving `RobertaForSequenceClassification` from `BertForSequenceClassification`.

## Test Plan

Added `test_xlm_roberta_models` to `tests/models/language/pooling/test_token_classification.py`, following the existing `test_bert_models`/`test_modernbert_models` pattern: loads `Davlan/xlm-roberta-base-ner-hrl` in both vLLM and HF Transformers, runs `token_classify` / `AutoModelForTokenClassification`, and asserts the per-token softmax logits match within tolerance.

Also registered `Davlan/xlm-roberta-base-ner-hrl` (XLM-RoBERTa) and `Jean-Baptiste/roberta-large-ner-english` (plain RoBERTa) in `tests/models/registry.py` for the standard model-registry consistency tests.

```
pytest tests/models/language/pooling/test_token_classification.py -k xlm_roberta
```

## Test Result

`ruff check` / `ruff format` pass on all changed files. Full model test requires downloading the HF checkpoints and was not run in this environment; relying on CI plus the fact that the implementation mirrors two already-tested code paths (`BertForTokenClassification`, `RobertaForSequenceClassification`) almost verbatim.

## Docs

Added both new architectures to the supported-models table in `docs/models/pooling_models/token_classify.md`.

### CODE REVIEW

**Comment by krishy91** [comment]
can someone mark the PR as ready?

**Review by noooop — APPROVED** [review]
Thanks for your contribution.
